### DISCUSSION TITLE
Android T (SDK 33) support?

### DISCUSSION THREAD

**Discussion by hoisie** [robolectric/robolectric#7338] [General]
A common question we receive is `when will Robolectric support Android T (SDK 33)`?

The answer is, as soon as Android T is released to AOSP, which should happen in the late July / early Aug 2022 timeframe. At that point we will push the code changes that will make Robolectric compatible with Android T. As soon as a Robolectric release is performed after this, it will support Android T.

Another question is `can Robolectric support pre-release versions of Android T(SDK 33)?

The answer is probably yes, but supporting prerelease SDKs is a lot more trouble than it's worth. There is a lot of churn in SDK code in prerelease SDKs (even up until the launch). Making these releases involve packaging prerelease framework jars and doing very frequent releases. This is a little too much overhead for the Robolectric team to handle given other priorities and projects.

Could there be features not testable for prereleases SDKs? Possibly, yes. However, from prior experience, for typical apps, generally not too much app code relies on prerelease SDK APIs, so it's usually not a huge deal to wait for T to be released.

**Comment by radixdev** [comment]
Would it be possible to to have the shadow manifest parser read a different target SDK version from the merged manifest during these beta periods (via some config)? 

A common issue during the beta periods is that the target version in the merged manifest says something like "android-T" and the robolectric parser fails to read it properly since it's a string. Would you be open to a pull request where this behavior can be configured to not crash in this way? It would also make some of these repeated questions (e.g. "when is Android XYZ support coming?") easier to handle since you can have your code on a preview version yet still run your tests (on an older target).

**Comment by ninniuz** [comment]
@hoisie since `Android T (SDK 33)` has been released to AOSP on Aug 15th, do you have any plan to share with us on Robolectric's support?

**Comment by utzcoz** [comment]
Android T supporting has been integrated into Robolectric 4.9.
